easybuild/easyblocks/a/amber.py

```diff
@@ -121,13 +121,20 @@ def install_step(self):
 
         # define environment variables for MPI, BLAS/LAPACK & dependencies
         mklroot = get_software_root('imkl')
+        openblasroot = get_software_root('OpenBLAS')
         if mklroot:
             env.setvar('MKL_HOME', mklroot)
+        elif openblasroot:
+            env.setvar('GOTO', os.environ['LIBLAPACK'])
```

Comment:
`os.getenv('LIBLAPACK', 'LIBLAPACK_NOT_SET')` may make more sense

In theory, you could list `OpenBLAS` as a dependency and build with an MPI toolchain like `gompi`, which would make the current code crash hard because `$LIBLAPACK` is not defined.

Reply:
Also, should we complain in an `else` that no LAPACK library was found?

Building without it is probably going to end in tears performance-wise?

Reply:
Can't do "-lopenblas -lgfortran" (which LIBLAPACK is with foss) since the openblas could have been built by another compiler, like intel or pgi and in that case -lgfortran is wrong.
And doing a list of all the possible combos is just wrong.
But technically speaking, openblas does, or at least used to , not depend on fortran libs, so LIBLAPACK being -lopenblas -lgfortran is wrong i think. And one could then do just -lopenblas here.

So, at the moment, we really have to rely on LIBLAPACK giving us the correct set of libraries. (And that applies to all other easyblocks/easyconfigs really). But complaining if LIBLAPACK isn't set is a good thing to do.

Reply:
@akesandgren Wait, `os.getenv('LIBLAPACK')` and `os.environ['LIBLAPACK']` are both going to give you the value for `$LIBLAPACK`...

Or are you just pointing out an issue in case `OpenBLAS` is combined with non-GCC compilers?

If so, please open an issue in the framework repository to track that.

That's orthogonal to this though?

Reply:
The issue is OpenBLAS built with non-GCC toolchains and probably that GCC-toolchain should define LIBLAPACK as just -lopenblas in this case, but i have too take a second look at how openblas is being built at the moment. It used to be the case that openblas was fortran-free. This might have changed with later versions and must be verified.

So orthogonal to the matter at hand yes.